src/Kestrel.Core/Internal/Http2/Http2Connection.cs

```diff
@@ -372,6 +373,12 @@ private Task ProcessDataFrameAsync()
                     throw new Http2ConnectionErrorException(CoreStrings.FormatHttp2ErrorStreamHalfClosedRemote(_incomingFrame.Type, stream.StreamId), Http2ErrorCode.STREAM_CLOSED);
                 }
 
+                if (stream.IsAborted)
+                {
+                    // Ignore data frames for aborted streams.
```

Comment:
This prevents data frames from being counted towards connection-level input flow control. I think this could lead to mismatched connection window size accounting by the client and server.